File: client/src/lib/types/conversation.ts
import type { Review } from './review';
import type { Account } from './account';
import type { Message } from './message';

export interface Conversation {
	id: string; // TODO UUID
	state: 'Accepted' | 'Quoted' | 'Rejected';
	created_at: Date;
}

export interface ConversationDTO {
	conversation: Conversation;
	messages: Message[];
	partner: Account;
  review: Review;
}

File: server/entity/src/dto/conversation.rs
use sea_orm::prelude::Uuid;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct ConversationDTO {
    pub conversation: crate::conversation::Model,
    pub partner: Option<crate::account::Model>,
    pub messages: Option<Vec<crate::message::Model>>,
    pub review: Option<Option<crate::review::Model>>,
}

#[derive(Debug, Deserialize)]
pub struct CreateConversationDTO {
    pub partner_id: Uuid,
}
